### ISSUE TITLE
Performance of textarea

### ISSUE THREAD

**Issue by leejack1000** [charmbracelet/bubbles#301]
Can the performance of textarea be optimized?
If I typed texts in textarea, it take a lot of cpu resources, and getting lag for a while.

**Comment by maaslalani** [comment]
Hey @leejack1000, this is something we are looking into improving at some point. We want to use a rope implementation to reduce memory usage and speed up the textarea.

**Comment by leejack1000** [comment]
Thanks @maaslalani, it's glad to hear this. Looking forward to seeing the results soon.

**Comment by mrusme** [comment]
@leejack1000 just in case you're looking for a quickfix to this, here's what I'm doing:

I have borrowed the code from [this PR](https://github.com/charmbracelet/lipgloss/pull/102) so that I can use the `PlaceOverlay` function in my project. Then I went ahead and called the `View` function that calls all the other `.View()` functions, from within the UI's `Update` function, and stored the output in a `Model` variable. Inside the `View` function (that calls all other `View` functions) I then check whether the `textarea` is focused and, if so, only call its `View` function and no other components' `View` function. I then utilize the `PlaceOverlay` function to place the updated textarea view on top of the cached view of the whole screen.

Long story short, this way it's possible to get pretty decent performance out of the textarea without the need to modify any bubbles or bubble tea itself. As soon as the texarea becomes active, only the cached view of the whole screen is being used and only the textarea gets `.View()` calls. It's sort of a poor-man's partial UI refresh.

**Comment by mrusme** [comment]
Having made the code public today I can now [link to what I've described above](https://github.com/mrusme/neonmodem/blob/08d9de64e025bbfef188d46eac2d77571ea9ca4b/ui/windows/postcreate/view.go#L22), in case anyone comes across the same issue and is looking for workaround.

**Comment by leejack1000** [comment]
> Having made the code public today I can now [link to what I've described above](https://github.com/mrusme/neonmodem/blob/08d9de64e025bbfef188d46eac2d77571ea9ca4b/ui/windows/postcreate/view.go#L22), in case anyone comes across the same issue and is looking for workaround.

Thank you for sharing.

**Comment by mikelorant** [comment]
All the pieces are now in place for this issue to be sorted out, just need some merges and dependencies updated.

Just to reiterate the performance improvements to expect, here is a comment I made on another issue.

> Just did some very quick benchmarks of my own app with pasting 1000 characters (14 lines) into a textarea of 72 characters wide:
>
> Before: 42s
> After: 3s
> 
> For context, no optimisations to my own app code at all. All these performance increases came from:
> - `bubbles`
> - `lipgloss`
> - `reflow`
> - `uniseg`

**Comment by bashbunni** [comment]
Hey, there hasn't been any activity on this issue in a long time and there have been a lot of changes since then. We are also coming out with a v2, currently living on `v2-exp` branch.

That said, if this is still an issue (especially in  `v2-exp`, please let us know.
